rclcpp/src/rclcpp/subscription_base.cpp

```diff
@@ -258,7 +269,16 @@ SubscriptionBase::get_message_type_support_handle() const
 bool
 SubscriptionBase::is_serialized() const
 {
-  return is_serialized_;
+  RCLCPP_WARN(
+    rclcpp::get_logger(
+      "rclcpp"), "is_serialized() is deprecated, use get_subscription_type() instead.");
```

Comment:
This isn't typically how we do deprecations, we do them with the `[[deprecated]]` annotation so it is a build-type warning.  This is going to be expensive, because you have to call into the logging machinery for every single message that is received.

Reply:
https://github.com/ros2/rclcpp/pull/2165/commits/331b7904ce75c222012e53c038e8a6d0ab3e7242